pkg/cloud/aws/aws.go

```diff
@@ -63,9 +67,11 @@ type awsProvider struct {
 	clusterName   string
 	instanceType  string
 	gateways      int
+	cloudPrepare  cpapi.Cloud
 }
 
 func NewAWSProvider(
+	restConfig *rest.Config,
```

Comment:
Why is this passed in? It looks unused. `rest.Config` is problematic for unit tests in terms of mocking so would be ideal if we can avoid passing it in code we want to unit test.

Reply:
Ah yes, thanks, it was needed with a previous version of cloud-prepare (with the MachineSetDeployer) but is no longer necessary.